src/talk/renderer/Viewer/ViewerHandlerText.vue

```diff
@@ -0,0 +1,182 @@
+  -->
+
+<script setup>
+import { computed, ref } from 'vue'
+import { t } from '@nextcloud/l10n'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+import NcEmptyContent from '@nextcloud/vue/dist/Components/NcEmptyContent.js'
+import NcRichText from '@nextcloud/vue/dist/Components/NcRichText.js'
+import IconCropPortrait from 'vue-material-design-icons/NoteTextOutline.vue'
+import IconCropLandscape from 'vue-material-design-icons/FileDocumentOutline.vue'
+import IconContentCopy from 'vue-material-design-icons/ContentCopy.vue'
+import IconFileOutline from 'vue-material-design-icons/FileOutline.vue'
+import ViewerHandlerBase from './ViewerHandlerBase.vue'
+import { useFileContent } from './viewer.composables.ts'
+import { toRef } from '@vueuse/core'
+
+const props = defineProps({
+	file: {
+		type: Object,
+		required: true,
+	},
+})
+
+const format = computed(() => {
+	const mimeToFormat = {
+		'text/markdown': 'md',
+		'text/plain': 'txt',
+	}
+	return mimeToFormat[props.file.mime] ?? 'code'
+})
+
+const layout = ref('compact')
+
+const {
+	content,
+	loading,
+	error,
+} = useFileContent(toRef(() => props.file.filename), 'text')
+</script>
+
+<template>
+	<ViewerHandlerBase :loading="loading" :error="error" :error-description="error">
+		<template #default>
+			<div class="viewer-text">
+				<fieldset :aria-label="t('talk_desktop', 'Controls')" class="viewer-text__controls">
+					<fieldset class="viewer-text__layout-switch" :aria-label="t('talk_desktop', 'Layout')">
+						<NcCheckboxRadioSwitch :checked.sync="layout"
+							:aria-label="t('talk_desktop', 'Compact')"
+							value="compact"
+							type="radio"
+							name="layout"
+							button-variant
+							button-variant-grouped="horizontal">
+							<template #icon>
+								<IconCropPortrait :size="20" />
+							</template>
+						</NcCheckboxRadioSwitch>
+						<NcCheckboxRadioSwitch :checked.sync="layout"
+							:aria-label="t('talk_desktop', 'Wide')"
+							value="wide"
+							type="radio"
+							name="layout"
+							button-variant
+							button-variant-grouped="horizontal">
+							<template #icon>
+								<IconCropLandscape :size="20" />
+							</template>
+						</NcCheckboxRadioSwitch>
+					</fieldset>
+					<NcButton :aria-label="t('talk_desktop', 'Copy content')"
+						:title="t('talk_desktop', 'Copy content')"
+						type="tertiary">
+						<template #icon>
+							<IconContentCopy :size="20" />
+						</template>
+					</NcButton>
+				</fieldset>
+
+				<div class="viewer-text__scroller">
+					<div v-if="content"
+						:aria-label="t('talk_desktop', 'Read-only text file content')"
+						class="viewer-text__content"
+						:class="[
+							`viewer-text__content--${format}`, {
+								'viewer-text__content--compact': layout === 'compact',
+							}]"
+						contenteditable
+						spellcheck="false"
+						@beforeinput.prevent>
```

Comment:
Is that needed, if we have read-only?

Reply:
We don't have readonly and it is not supported on `[contenteditable]`. So we have to emulate readonly...